src/components/Map/Map.tsx

```diff
@@ -43,17 +44,27 @@ export type MapProps = {
   entityOptions?: MapEntityOptions;
 };
 
-export default function Map(props: MapProps): JSX.Element {
+export default function Map(props: MapProps): JSX.Element | null {
   const { token, onTokenExpired, options, popupRenderer, mapId, style, bannerMessage, entityOptions } = props;
   const [geoData, setGeoData] = useState<any[]>();
   const [layerIds, setLayerIds] = useState<string[]>([]);
   const [popupInfo, setPopupInfo] = useState<PopupInfo | null>(null);
   const [deferredHighlightEntity, setDeferredHighlightEntity] = useState<MapEntityId | undefined>();
   const [deferredFocusEntity, setDeferredFocusEntity] = useState<MapEntityId | undefined>();
   const mapRef = useRef(null);
+  const containerRef = useRef(null);
   const hoverStateId: { [key: string]: number | undefined } = useMemo(() => ({}), []);
   const selectStateId: { [key: string]: number | undefined } = useMemo(() => ({}), []);
   const highlightStateId: { [key: string]: number | undefined } = useMemo(() => ({}), []);
+  const [firstVisible, setFirstVisible] = useState(false);
+  const visible = useIsVisible(containerRef);
+
+  useEffect(() => {
+    // `firstVisible` detects when the box containing the map is first visible in the viewport. The map should only be
+    // rendered if `firstVisible` is true. This accounts for cases in which the map is initially rendered hidden, and
+    // is improperly resized when it first becomes visible.
+    setFirstVisible((fv) => fv || visible);
```

Comment:
Here's how the map initially renders when it starts out hidden and is then shown:
![image](https://github.com/terraware/terraware-web/assets/114949086/1b958f2e-29ad-4b03-8df6-b8e259d13f3e)